sdk/python/feast/infra/materialization/batch_materialization_engine.py

```diff
@@ -0,0 +1,87 @@
+import dataclasses
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Callable, List, Optional, Sequence, Union
+
+from tqdm import tqdm
+
+from feast.batch_feature_view import BatchFeatureView
+from feast.entity import Entity
+from feast.feature_view import FeatureView
+from feast.infra.offline_stores.offline_store import OfflineStore
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.repo_config import RepoConfig
+from feast.stream_feature_view import StreamFeatureView
+
+
+@dataclasses.dataclass
```

Comment:
ideally would have docstrings for these new interfaces, but I'm fine with incrementally adding them as we actually implement them

Reply:
yeah I'll add stubs